test_main.py

```diff
@@ -0,0 +1,36 @@
+"""
+This is the test !!!
+"""
+import unittest
+from app import get_guess_computer, get_guess_Player, mainFunction
+from unittest.mock import patch
+import io
+import random
+
+class TestMainCode(unittest.TestCase):
+    def test_get_computer(self):
+        with patch('sys.stdout', new= io.StringIO()) as fake_out:
+            result = get_guess_computer()
+            self.assertIsInstance(result, int)
+            self.assertGreaterEqual(result, 1)
+            self.assertLessEqual(result, 100)
+            self.assertEqual(fake_out.getvalue(), '')
+
+    def test_get_player(self):
+        with patch('sys.stdout', new= io.StringIO()) as fake_out:
+            result= get_guess_Player()
+            self.assertIsInstance(result, int)
+            self.assertGreaterEqual(result, 1)
+            self.assertLessEqual(result, 100)
+            self.assertEqual(fake_out.getvalue(), '')
+
+    def test_main_function(self):
+      secret_number = random.randint(1, 100)
+      with patch('sys.stdout', new=io.StringIO()) as fake_out:
+        mainFunction(secret_number)
+        print("Captured Output:", fake_out.getvalue())
+        self.assertEqual(fake_out.getvalue(), '')
```

Comment:
I don't understand this test, it seems like you are asserting things are getting printed. Think first about what you might want to test about your main -
1. the guess functions are called
2. if a guess is correct the game ends
3. if no one guesses right the loop continues (the guess functions are called again)
...

You can spy on functions and assert they are called with arguments with patch to mock
https://realpython.com/python-mock-library/#assertions-and-inspection
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_count

Reply:
Im gona check this and reformulate my tests, honestly I dont undestand complety stdout I just was tryng anything for my code works